lib/log.js

```diff
@@ -178,25 +178,59 @@ Lug.prototype.activityEvent = function (event, request, data) {
     return P.resolve()
   }
 
+  // log a flowEvent for all activityEvents
+  this.flowEvent(event, request)
+
+  var info = {
+    event: event
+  }
+
+  if (request.headers['user-agent']) {
+    info.userAgent = request.headers['user-agent']
+  }
+
+  optionallySetService(info, request)
+
+  Object.keys(data).forEach(function (key) {
+    info[key] = data[key]
+  })
+
+  this.logger.info('activityEvent', info)
+  this.statsd.write(info)
+
+  return P.resolve()
+}
+
+// Log a flow metrics event.
+// These events help understand the user flow.
+
+Lug.prototype.flowEvent = function (event, request) {
   var self = this
 
+  if (! event) {
+    this.error({ op: 'log.flowEvent', missingEvent: true })
+    return P.resolve()
+  }
+
+  if (! request || ! request.headers) {
+    this.error({ op: 'log.flowEvent', event: event, missingHeaders: true })
```

Comment:
Probably bike-shedding here, but do we mind that `missingHeaders` is kind of inaccurate in the `! request` case? Would something like `badRequest` be more appropriate?